fingertip/plugins/cleanup.py

```diff
@@ -100,6 +101,39 @@ def tempfiles(older_than='4h', location=None):
                                       temp.AUTOREMOVE_PREFIX not in f))
 
 
+def saviour(expired_for='10d'):
+    if expired_for != 'all':
+        adjusted_time = time.time() - units.parse_time_interval(expired_for)
+    else:
+        adjusted_time = 0
+    root, dirs, files = next(os.walk(path.SAVIOUR + '/_'))
+    for d in (os.path.join(root, x) for x in dirs):
+        lock_path = os.path.join(root, '.' + os.path.basename(d) + '-lock')
+        lock = fasteners.process_lock.InterProcessLock(lock_path)
+        lock.acquire()
+        time_file_path = os.path.join(d, ".last_mirrored")
+        data_path = os.path.join(d, "data")
+        if os.path.exists(time_file_path):
+            remove = _time(time_file_path) <= adjusted_time
+        elif os.path.exists(data_path):
+            remove = _time(data_path) <= adjusted_time
```

Comment:
not sure I'm comfortable having this comparison if rsync sets folder date to some ancient times